common/src/main/java/org/apache/celeborn/common/write/SlowStartPushStrategy.java

```diff
@@ -0,0 +1,170 @@
+
+public class SlowStartPushStrategy extends PushStrategy {
+
+  protected static class CongestControlContext {
+    private final AtomicInteger currentMaxReqsInFlight;
+
+    // Indicate the number of congested times even after the in flight requests reduced to 1
+    private final AtomicInteger continueCongestedNumber;
+    private int congestionAvoidanceFlag;
+    private int reqsInFlightBlockThreshold;
+
+    public CongestControlContext(int reqsInFlightBlockThreshold) {
+      this.currentMaxReqsInFlight = new AtomicInteger(1);
+      this.continueCongestedNumber = new AtomicInteger(0);
+      this.congestionAvoidanceFlag = 0;
+      this.reqsInFlightBlockThreshold = reqsInFlightBlockThreshold;
+    }
+
+    public synchronized void increaseCurrentMaxReqs() {
+      continueCongestedNumber.set(0);
+      if (currentMaxReqsInFlight.get() >= reqsInFlightBlockThreshold) {
+        // Congestion avoidance
+        congestionAvoidanceFlag++;
+        if (congestionAvoidanceFlag >= currentMaxReqsInFlight.get()) {
+          currentMaxReqsInFlight.incrementAndGet();
+          congestionAvoidanceFlag = 0;
+        }
+      } else {
+        // Slow start
+        currentMaxReqsInFlight.incrementAndGet();
+      }
+    }
+
+    public synchronized void decreaseCurrentMaxReqs() {
+      if (currentMaxReqsInFlight.get() <= 1) {
+        currentMaxReqsInFlight.set(1);
+        continueCongestedNumber.incrementAndGet();
+      } else {
+        currentMaxReqsInFlight.updateAndGet(pre -> pre / 2);
+      }
+      reqsInFlightBlockThreshold = currentMaxReqsInFlight.get();
+      congestionAvoidanceFlag = 0;
+    }
+
+    public int getCurrentMaxReqsInFlight() {
+      return currentMaxReqsInFlight.get();
+    }
+
+    public int getContinueCongestedNumber() {
+      return continueCongestedNumber.get();
+    }
+  }
+
+  private static final Logger logger = LoggerFactory.getLogger(SlowStartPushStrategy.class);
+
+  private final int maxInFlight;
+  private final long initialSleepMills;
+  private final long maxSleepMills;
+  private final ConcurrentHashMap<String, CongestControlContext> congestControlInfoPerAddress;
+
+  public SlowStartPushStrategy(CelebornConf conf) {
+    super(conf);
+    this.maxInFlight = conf.pushMaxReqsInFlight();
+    this.initialSleepMills = conf.pushSlowStartInitialSleepTime();
+    this.maxSleepMills = conf.pushSlowStartMaxSleepMills();
+    this.congestControlInfoPerAddress = new ConcurrentHashMap<>();
+  }
+
+  @VisibleForTesting
+  protected CongestControlContext getCongestControlContextByAddress(String hostAndPushPort) {
+    return congestControlInfoPerAddress.computeIfAbsent(
+        hostAndPushPort, host -> new CongestControlContext(maxInFlight));
+  }
+
+  /**
+   * If `pushDataSlowStart` is enabled, will increase `currentMaxReqsInFlight` gradually to meet the
+   * max push speed.
+   *
+   * <p>1. slow start period: every RTT period, `currentMaxReqsInFlight` is doubled.
```

Comment:
I don't see where currentMaxReqsInFlight is doubled, should it be line 58?

Reply:
Yea, from @AngersZhuuuu's initial idea, here we define one RTT period means one batch of push data requests(batch size -> `currentMaxReqsInFlight`). So we plus 1 for every request, and for one batch request, the next batch will be increased to currentMaxReqsInFlight * currentMaxReqsInFlight, which is doubled.

e.g.
current currentMaxReqsInFlight is 1, the next maxReqs is 2
current currentMaxReqsInFlight is 2, the next maxReqs is 4, etc...

Reply:
> I don't see where currentMaxReqsInFlight is doubled, should it be line 58?

In this method `CongestControlContext .increaseCurrentMaxReqs()`